### DISCUSSION TITLE
window.Alpine is not optional

### DISCUSSION THREAD

**Discussion by assertchris** [alpinejs/alpine#2314] [1. Help] [answered]
❤️ your work!

Refactoring an older codebase, and one of the things included in the refactor is upgrading from v2 to v3.

## Context
1. Before `Alpine.data`; we were using `window.MyComponent = () => {...}`, and `window.Alpine = Alpine`.
2. One of the first things I did was try to reduce the number of `window.MyComponent = () => {...}`, and use `Alpine.data`.
3. The docs say `window.Alpine = Alpine` is optional¹ but I was seeing errors.
4. A source-dive shows a number of places where `window.Alpine` is assumed.

## Potential solutions
1. Remove the note about `window.Alpine` being optional. Perhaps even make it clear that it's not, and shouldn't be skipped.
2. Find a way not to depend on `window.Alpine` inside Alpine.

> ¹ <img width="756" alt="Screenshot 2021-11-04 at 06 51 07" src="https://user-images.githubusercontent.com/200609/140260160-6acf51b0-9ed8-4c79-879f-d0e51343ad1c.png">

**Comment by SimoTod** [comment]
It's technically optional unless you start using the Alpine variable in other scripts. Can you post an example where the source code uses window.Alpine?

**Comment by assertchris** [comment, marked as answer]
A few spots over here: https://github.com/alpinejs/alpine/search?q=window.Alpine

**Reply by danddanddand** [reply]
Those were almost all just CDN build, a test and documentation and a plug-in that isn't in core?

**Reply by assertchris** [reply]
Hmmm, how about this one? https://github.com/alpinejs/alpine/blob/4d319afbfe0e274d552d3f6e968aca8899c05676/packages/morph/src/morph.js#L276

**Reply by SimoTod** [reply]
that package hasn't been released yet. plugins have access to Alpine so they don't need window.Alpine. It's just old code.

**Reply by assertchris** [reply]
Ok, great! The original issue was for a window.Alpine.version check; but I can't figure out how I was seeing that since all recent tagged versions have the code in question commented out. If it's not in any core code then consider this a non-issue. Thanks for the clarification.
